core/services/directrequest/delegate.go

```diff
@@ -179,8 +179,8 @@ func (l *listener) run() {
 
 func (l *listener) handleReceivedLogs() {
 	for {
-		i := l.mbLogs.Retrieve()
-		if i == nil {
+		i, exists := l.mbLogs.Retrieve()
+		if !exists {
```

Comment:
What does it actually mean if the mailbox is empty?

Reply:
it would be very rare - it's the case i mentioned in the description. It just means that all items were already picked and there was one excessive iteration of the loop